### PR TITLE
Add help subcommand

### PR DESCRIPTION
**Pull request by sirnewton01**
Let's add a help subcommand and clean up the help. Also, the standard git outputs usages on stdout, not stderr.

### CODE REVIEW

**Comment by sirnewton01** [comment]
Ah, it's intent based. If you are seeking help then stdout. Otherwise, if it is a usage failure then stderr. Okay, I'll try to see how to adjust the design.

File: .travis.yml [hunk 1]
```diff
@@ -4,4 +4,5 @@ go:
   - "1.10.x"
   - master
 
-
```
Comment (sirnewton01) [inline .travis.yml]: I kept forgetting to format my source code before opening the pull requests. This travis CI addition checks that everything is formatted properly. I can turn this off if you disagree with the philosophy.

Comment (driusan) [reply .travis.yml]: I'm not sure if it'll cause superfluous failures with different go versions, but if it does we can take it out later.

You overrode the default script section though, so it's not running `go test -v ./...` anymore unless you explicitly add it back.

File: cmd/apply.go [hunk 2]
```diff
@@ -3,14 +3,17 @@ package cmd
...
 	"flag"
 	"fmt"
+	"os"
 
 	"github.com/driusan/dgit/git"
 )
 
 func Apply(c *git.Client, args []string) error {
 	flags := flag.NewFlagSet("apply", flag.ExitOnError)
+	flags.SetOutput(os.Stdout)
```
Comment (driusan) [inline cmd/apply.go]: Why do some of these use `os.Stdout` and other ones use `flag.CommandLine.Output()`?

Comment (sirnewton01) [reply cmd/apply.go]: This was left over from a previous commit. Thanks, I have fixed those.

**Pull request merged**